regcomp.h uses C11 feature (anonymous union/struct)

Since commit 17e3e02ad120eabda2bdb6c297a70d53294437ef, regcomp.h makes use of anonymous unions and structs, e.g.:
```c
struct regnode_anyofhs { /* Constructed this way to keep the string aligned. */
    U8	str_len;
    U8  type;
    U16 next_off;
    union {
        U32 arg1u;
        I32 arg1i;
        struct {
            U16 arg1a;
            U16 arg1b;
        };
    };
    char string[1];
};
```
This feature is only available in C11 (and previously as a GNU extension): https://en.cppreference.com/w/c/language/union

I don't think perl wants to require a C11 compiler yet, so these anonymous members should get names.

In gcc there is the `-Wc99-c11-compat` flag. It would detect this case:
```
../../regcomp.h:188:10: warning: ISO C99 doesn’t support unnamed structs/unions [-Wc99-c11-compat]
  188 |         };               
      |          ^   
```
But it would also complain about
```
../../config.h:5190:27: warning: ISO C99 does not support ‘_Thread_local’ [-Wc99-c11-compat]                                                                                                                                                  
 5190 | #define PERL_THREAD_LOCAL _Thread_local /**/
      |                           ^~~~~~~~~~~~~                                                                        
../../thread.h:400:8: note: in expansion of macro ‘PERL_THREAD_LOCAL’
  400 | extern PERL_THREAD_LOCAL void *PL_current_context;
      |        ^~~~~~~~~~~~~~~~~                                                                                       
```

`-Wpedantic` would also catch this, but it would complain about everything else, too:
```
In file included from perl.h:3299,                                                                                                                                                                                                            
                 from regcomp.c:134:                                                                                                                                                                                                          
cv.h: In function ‘CvNAME_HEK’:                                                                                                                                                                                                               
handy.h:99:26: warning: ISO C forbids braced-groups within expressions [-Wpedantic]                                                                                                                                                           
   99 | #  define MUTABLE_PTR(p) ({ void *p_ = (p); p_; })                                                                                                                                                                                    
      |                          ^                                                                                                                                                                                                            
cv.h:91:36: note: in expansion of macro ‘MUTABLE_PTR’                                                                                                                                                                                         
   91 | #define CvFLAGS(sv)       ((XPVCV*)MUTABLE_PTR(SvANY(sv)))->xcv_flags                                                                                                                                                                 
      |                                    ^~~~~~~~~~~                                                                                                                                                                                        
cv.h:218:34: note: in expansion of macro ‘CvFLAGS’                                                                                                                                                                                            
  218 | #define CvNAMED(cv)             (CvFLAGS(cv) & CVf_NAMED)                                                                                                                                                                             
      |                                  ^~~~~~~                                                                                                                                                                                              
cv.h:281:12: note: in expansion of macro ‘CvNAMED’                                                                                                                                                                                            
  281 |     return CvNAMED(sv)                                                                                                                                                                                                                
      |            ^~~~~~~                                                                                                                                                                                                                    
```

--- a/regcomp.c
+++ b/regcomp.c
@@ -11927,7 +11927,7 @@ S_optimize_regclass(pTHX_
         &&   *anyof_flags == 0
         &&   start[0] < (1 << ANYOFR_BASE_BITS)
         &&   end[0] - start[0]
-                < ((1U << (sizeof(((struct regnode_1 *)NULL)->arg1u)
+                < ((1U << (sizeof(ARG1u_LOC(NULL))
                                * CHARBITS - ANYOFR_BASE_BITS))))
 
     {

--- a/regcomp.h
+++ b/regcomp.h
@@ -331,7 +331,7 @@ struct regnode_charclass {
         U32 arg1u;
         struct {
             U16 arg1a;
-            U16 arg2a;
+            U16 arg1b;
         };
     };
     char bitmap[ANYOF_BITMAP_SIZE];	/* only compile-time */
@@ -347,7 +347,7 @@ struct regnode_charclass_posixl {
         U32 arg1u;
         struct {
             U16 arg1a;
-            U16 arg2a;
+            U16 arg1b;
         };
     };
     char bitmap[ANYOF_BITMAP_SIZE];		/* both compile-time ... */
@@ -376,7 +376,7 @@ struct regnode_ssc {
         U32 arg1u;
         struct {
             U16 arg1a;
-            U16 arg2a;
+            U16 arg1b;
         };
     };
     char bitmap[ANYOF_BITMAP_SIZE];	/* both compile-time ... */

--- a/regcomp.h
+++ b/regcomp.h
@@ -185,8 +185,8 @@ struct regnode_anyofhs { /* Constructed this way to keep the string aligned. */
         struct {
             U16 arg1a;
             U16 arg1b;
-        };
-    };
+        } hi_lo;
+    } arg1;
     char string[1];
 };
 
@@ -204,8 +204,8 @@ struct regnode_1 {
         struct {
             U16 arg1a;
             U16 arg1b;
-        };
-    };
+        } hi_lo;
+    } arg1;
 };
 
 /* Node whose argument is 'SV *'.  This needs to be used very carefully in
@@ -244,16 +244,16 @@ struct regnode_2 {
         struct {
             U16 arg1a;
             U16 arg1b;
-        };
-    };
+        } hi_lo;
+    } arg1;
     union {
         U32 arg2u;
         I32 arg2i;
         struct {
             U16 arg2a;
             U16 arg2b;
-        };
-    };
+        } hi_lo;
+    } arg2;
 };
 
 /* "Three Node" - similar to a regnode_2 but with space for an additional
@@ -274,24 +274,24 @@ struct regnode_3 {
         struct {
             U16 arg1a;
             U16 arg1b;
-        };
-    };
+        } hi_lo;
+    } arg1;
     union {
         I32 arg2i;
         U32 arg2u;
         struct {
             U16 arg2a;
             U16 arg2b;
-        };
-    };
+        } hi_lo;
+    } arg2;
     union {
         struct {
             U16 arg3a;
             U16 arg3b;
-        };
+        } hi_lo;
         I32 arg3i;
         U32 arg3u;
-    };
+    } arg3;
 };
 
 #define REGNODE_BBM_BITMAP_LEN                                                  \
@@ -332,8 +332,8 @@ struct regnode_charclass {
         struct {
             U16 arg1a;
             U16 arg1b;
-        };
-    };
+        } hi_lo;
+    } arg1;
     char bitmap[ANYOF_BITMAP_SIZE];	/* only compile-time */
 };
 
@@ -348,8 +348,8 @@ struct regnode_charclass_posixl {
         struct {
             U16 arg1a;
             U16 arg1b;
-        };
-    };
+        } hi_lo;
+    } arg1;
     char bitmap[ANYOF_BITMAP_SIZE];		/* both compile-time ... */
     U32 classflags;	                        /* and run-time */
 };
@@ -377,8 +377,8 @@ struct regnode_ssc {
         struct {
             U16 arg1a;
             U16 arg1b;
-        };
-    };
+        } hi_lo;
+    } arg1;
     char bitmap[ANYOF_BITMAP_SIZE];	/* both compile-time ... */
     U32 classflags;	                /* ... and run-time */
 
@@ -553,18 +553,18 @@ struct regnode_ssc {
 
 #define	NODE_ALIGN(node)
 #define ARGp_BYTES_LOC(p)  (((struct regnode_p *)p)->arg1_sv_ptr_bytes)
-#define ARG1u_LOC(p)    (((struct regnode_1 *)p)->arg1u)
-#define ARG1i_LOC(p)    (((struct regnode_1 *)p)->arg1i)
-#define ARG1a_LOC(p)    (((struct regnode_1 *)p)->arg1a)
-#define ARG1b_LOC(p)    (((struct regnode_1 *)p)->arg1b)
-#define ARG2u_LOC(p)    (((struct regnode_2 *)p)->arg2u)
-#define ARG2i_LOC(p)    (((struct regnode_2 *)p)->arg2i)
-#define ARG2a_LOC(p)    (((struct regnode_2 *)p)->arg2a)
-#define ARG2b_LOC(p)    (((struct regnode_2 *)p)->arg2b)
-#define ARG3u_LOC(p)    (((struct regnode_3 *)p)->arg3u)
-#define ARG3i_LOC(p)    (((struct regnode_3 *)p)->arg3i)
-#define ARG3a_LOC(p)    (((struct regnode_3 *)p)->arg3a)
-#define ARG3b_LOC(p)    (((struct regnode_3 *)p)->arg3b)
+#define ARG1u_LOC(p)    (((struct regnode_1 *)p)->arg1.arg1u)
+#define ARG1i_LOC(p)    (((struct regnode_1 *)p)->arg1.arg1i)
+#define ARG1a_LOC(p)    (((struct regnode_1 *)p)->arg1.hi_lo.arg1a)
+#define ARG1b_LOC(p)    (((struct regnode_1 *)p)->arg1.hi_lo.arg1b)
+#define ARG2u_LOC(p)    (((struct regnode_2 *)p)->arg2.arg2u)
+#define ARG2i_LOC(p)    (((struct regnode_2 *)p)->arg2.arg2i)
+#define ARG2a_LOC(p)    (((struct regnode_2 *)p)->arg2.hi_lo.arg2a)
+#define ARG2b_LOC(p)    (((struct regnode_2 *)p)->arg2.hi_lo.arg2b)
+#define ARG3u_LOC(p)    (((struct regnode_3 *)p)->arg3.arg3u)
+#define ARG3i_LOC(p)    (((struct regnode_3 *)p)->arg3.arg3i)
+#define ARG3a_LOC(p)    (((struct regnode_3 *)p)->arg3.hi_lo.arg3a)
+#define ARG3b_LOC(p)    (((struct regnode_3 *)p)->arg3.hi_lo.arg3b)
 
 /* These should no longer be used directly in most cases. Please use
  * the REGNODE_AFTER() macros instead. */

